TLM/TLM/UI/Helpers/UITriStateCheckbox.cs

```diff
@@ -0,0 +1,107 @@
+namespace TrafficManager.UI.Helpers {
+    using ColossalFramework.UI;
+    using System;
+    using TrafficManager.U;
+    using TrafficManager.Util;
+    using UnityEngine;
+
+    public class UITriStateCheckbox : UICheckBox {
+        public event PropertyChangedEventHandler<bool?> EventValueChanged;
+
+        public UIComponent FalseComponent;
+        public UIComponent NullComponent;
+        public UIComponent TrueComponent {
+            get => checkedBoxObject;
+            set => checkedBoxObject = value;
+        }
+
+        private bool? value_;
+        public bool? Value {
+            get => value_;
+            set {
+                if (value != value_) {
+                    value_ = value;
+                    OnValueChanged();
+                }
+            }
+        }
+
+        [Obsolete("use Value instead", error: true)]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.NamingRules", "SA1300:Element should begin with upper-case letter", Justification = "N/A")]
+        public new bool isChecked { get; set; }
```

Comment:
Why lower-case and why `new`?

Reply:
If it's brand new class why that property is even obsolete? It shouldn't be here.

Reply:
I am trying to mask the inherited `isChecked` property because it does not give the right value. `new` to hide warning

Reply:
but.. it may not work correctly, eg, if base checkbox class is using `isChecked` prop then this one with `new` won't be called since `new` is not the same as `override`. Normally you would have create new type instead of extending UIcheckbox, since checkbox does not support null value itself but here.. we don't need that.

Reply:
I mean its not full proof but its better than nothing. 
base class is calling `isChecked` but we don't care about that. I just don't want us to call be able to call `isChecked`.
I can remove it if you like.

Reply:
naah, it's ok. I get the idea